### PR TITLE
fix: panic when setting a zero interval for ticker

### PR DESCRIPTION
**Pull request by docmerlin**
###### Required for all non-trivial PRs
- [ ] Rebased/mergable
- [ ] Tests pass
- [ ] CHANGELOG.md updated

Closes https://github.com/influxdata/kapacitor/issues/2281

### CODE REVIEW

File: barrier.go [hunk 1]
```diff
@@ -80,7 +80,7 @@ func (n *BarrierNode) newBarrier(group edge.GroupInfo, first edge.PointMeta) (ed
 		)
 		return periodicBarrier, periodicBarrier.Stop, nil
 	default:
-		return nil, nil, errors.New("unreachable code, barrier node should have non-zero idle or non-zero period")
+		return nil, nil, errors.New("unreachable code, barrier node should have positive idle or positive period")
```
Comment (wolffcm) [inline barrier.go:83]: I'm curious, what is the behavior supposed to be if `period` and `idle` are both set? Is that possible?

Comment (wolffcm) [reply barrier.go:83]: The docs say it's based on one of them. I wonder if you should return an error if both are set?

File: pipeline/json.go [hunk 2]
```diff
@@ -447,6 +448,9 @@ func unmarshalStats(data []byte, parents []Node, typ TypeOf) (Node, error) {
 	}
 	child := nodeParent.Stats(0)
 	err := json.Unmarshal(data, child)
+	if child.Interval == 0 {
+		return nil, errors.New("zero is an invalid stats interval")
```
Comment (wolffcm) [inline pipeline/json.go:452]: This code is for serializing kapacitor pipelines, isn't it? Doesn't this check belong in `stats.go`?

File: barrier.go (hunk 1 above)
Comment (docmerlin) [reply barrier.go:83]: That is probably a good idea.  I think it is outside the scope of this PR though.

Comment (wolffcm) [reply barrier.go:83]: Yeah, that makes sense.

File: pipeline/json.go (hunk 2 above)
Comment (docmerlin) [reply pipeline/json.go:452]: I added one to stats.go

**Pull request merged**